### PR TITLE
Move network configuration into vault-on-gke composition

### PR DESCRIPTION
**Pull request by ezgidemirel**
Signed-off-by: ezgidemirel <ezgidemirel91@gmail.com>

### Description of your changes
This change is moving network configuration to the vault-on-gke composition. 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).

### How has this code been tested
After creating a package and pushing it to the Upbound registry, I created a vault-on-gke instance from Upbound Cloud and verified that Vault deployment on the GKE cluster is working as expected.

### CODE REVIEW

File: configuration/cluster/gke/composition.yaml [hunk 1]
```diff
@@ -40,9 +40,9 @@ spec:
         - fromFieldPath: spec.writeConnectionSecretToRef.namespace
           toFieldPath: spec.writeConnectionSecretToRef.namespace
         # Create this cluster in the Network and Subnetwork referenced by network ID
-        - fromFieldPath: "spec.parameters.networkRef.id"
+        - fromFieldPath: "spec.id"
```
Comment (turkenh) [inline configuration/cluster/gke/composition.yaml:43]: Did had a chance to check/figure out whether we could remove `id` field of top level XR spec in favor of `metadata.name` ?

**Review by turkenh — APPROVED** [review]
Thank you @ezgidemirel, looking good!

File: configuration/cluster/gke/composition.yaml (hunk 1 above)
Comment (ezgidemirel) [reply configuration/cluster/gke/composition.yaml:43]: My understanding might be wrong, but I assumed it's required for Upbound Cloud UI. If you think it's not related, I'm happy to give it a try

Comment (turkenh) [reply configuration/cluster/gke/composition.yaml:43]: I don't think it is that important ATM but would be nice to track with a simple issue not to forget.

**Pull request merged**